qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -149,26 +147,46 @@ class AddIntoPhaseGrad(GateWithRegisters, cirq.ArithmeticGate):
         [Compilation of Fault-Tolerant Quantum Heuristics for Combinatorial Optimization]
         (https://arxiv.org/abs/2007.07391), Appendix A: Addition for controlled rotations
     """
-    inp_bitsize: int
+    x_bitsize: int
     phase_bitsize: int
+    right_shift: int = 0
+    sign: int = +1
+
+    def pretty_name(self) -> str:
+        sign = '+' if self.sign > 0 else '-'
+        return f'pg{sign}=x>>{self.right_shift}' if self.right_shift else f'pg{sign}=x'
 
     @cached_property
     def signature(self) -> 'Signature':
         return Signature.build_from_dtypes(
-            x=QUInt(self.inp_bitsize), phase_grad=QFxp(self.phase_bitsize, self.phase_bitsize)
+            x=QFxp(self.x_bitsize, self.x_bitsize, signed=False),
+            phase_grad=QFxp(self.phase_bitsize, self.phase_bitsize, signed=False),
         )
 
+    @cached_property
+    def phase_dtype(self) -> QFxp:
+        return QFxp(self.phase_bitsize, self.phase_bitsize, signed=False)
+
     def registers(self) -> Sequence[Union[int, Sequence[int]]]:
-        return [2] * self.inp_bitsize, [2] * self.phase_bitsize
+        return [2] * self.x_bitsize, [2] * self.phase_bitsize
 
     def with_registers(self, *new_registers: Union[int, Sequence[int]]):
         raise NotImplementedError("not needed.")
 
-    def apply(self, x, phase_grad) -> Union[int, Iterable[int]]:
-        return x, phase_grad + x
+    @cached_method
+    def scaled_val(self, x: int) -> int:
+        """Computes `phase_grad + x` using fixed point arithmetic."""
+        x_width = self.x_bitsize + self.right_shift
+        x_fxp = _fxp(x / 2**x_width, x_width).like(_fxp(0, self.phase_bitsize)).astype(float)
```

Comment:
am i reading this line right:

1. x / 2**x_width of width x_width 
2. Convert to a new fxp of width phase_bitsize
3. convert to float.

Why is step 2 required?